### PR TITLE
[Caffe2] Add in_place test for binary ops

### PR DESCRIPTION
**Pull request by xiaomengy**

### CODE REVIEW

File: caffe2/python/operator_test/elementwise_ops_test.py [hunk 1]
```diff
@@ -376,49 +378,86 @@ def ref(A, B):
...
            t=st.integers(0, 5), **hu.gcs)
     def test_sub(self, n, m, k, t, gc, dc):
         self._test_binary_op("Sub", np.subtract, n, m,
                              k, t, -0.5, True, gc, dc)
+        self._test_binary_op_in_place(
+            "Sub", np.subtract, n, m,  -0.5, True, True, gc, dc)
 
     @given(n=st.integers(0, 5), m=st.integers(0, 5), k=st.integers(0, 5),
-           t=st.integers(0, 5), **hu.gcs)
+           t=st.integers(0, 5), **hu.gcs_cpu_only)
```
Comment (houseroad) [inline caffe2/python/operator_test/elementwise_ops_test.py]: Something is wrong with GPU's `Mul` implementation?

**Review by houseroad — APPROVED** [review]
Overall LGTM, however, not sure why Mul is only tested on CPU

File: caffe2/python/operator_test/elementwise_ops_test.py (hunk 1 above)
Comment (BIT-silence) [reply caffe2/python/operator_test/elementwise_ops_test.py]: Done

**Pull request closed without merging**